force_gromacs/data_sources/molecule/molecule_data_source.py

```diff
@@ -1,51 +1,61 @@
+import copy
+
 from force_bdss.api import BaseDataSource, DataValue, Slot
 
-from force_gromacs.io.gromacs_topology_reader import (
-    GromacsTopologyReader
+from force_gromacs.data_sources.molecule import (
+    Molecule
 )
 
-from .molecule import Molecule
-
 
 class MoleculeDataSource(BaseDataSource):
-    """Class takes in all data required to define each
-    separate molecule molecule in a Gromacs experiment. Gromacs topology
-    files must be included for all species, however coordinate files
-    are not necessary for atoms or molecules represented by single beads.
+    """Class that collates a set of molecular `Fragment` instances into
+    a single chemical `Molecule`.
     """
 
-    # --------------------
-    #  Regular Attributes
-    # --------------------
+    def _make_local_parameter_copy(self, parameters):
```

Comment:
This is important, since multiple `MoleculeSatSources` can share the same `FragmentDataSource` and will try to access its `number` attribute.